Library/Formula/gdal.rb

```diff
@@ -284,6 +290,15 @@ def install
       bin.install Dir['scripts/*']
     end
 
+    if build.with? "swig-java"
+      cd 'swig/java' do
+        inreplace "java.opt", "linux", "darwin"
```

Comment:
The default java.opt (config file) is for linux but the build can support macosx

The in replace changes this

    JAVA_INCLUDE=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux

to this

    JAVA_INCLUDE=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/darwin